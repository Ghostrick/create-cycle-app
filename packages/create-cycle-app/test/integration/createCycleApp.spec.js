const fs = require('fs-extra')
const path = require('path')
const appName = path.resolve(__dirname, 'MYCYCLEAPP')
const spawn = require('cross-spawn')

describe('create-cycle-app', () => {
  afterEach(() => {
    fs.removeSync(appName)
  })
  beforeEach((done) => {
    spawn.sync('node', [
      path.resolve(__dirname, '../../index.js'),
      appName
    ])
    done()
  })
  test('', () => {
    const dir = fs.readdirSync(appName)
    expect(dir).toMatchObject([
      '.gitignore',
      'node_modules',
      'package.json',
      'public',
      'src'
    ])
    expect(fs.readFileSync(path.join(appName, 'package.json'), 'UTF8')).toMatchSnapshot()
  })
})
