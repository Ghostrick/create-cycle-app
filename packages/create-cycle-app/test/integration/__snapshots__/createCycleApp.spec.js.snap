exports[`create-cycle-app  1`] = `
"{
  \"name\": \"MYCYCLEAPP\",
  \"version\": \"0.1.0\",
  \"private\": true,
  \"devDependencies\": {
    \"cycle-scripts\": \"1.0.2\"
  },
  \"dependencies\": {},
  \"scripts\": {
    \"start\": \"cycle-scripts start\",
    \"test\": \"cycle-scripts test\",
    \"build\": \"cycle-scripts build\",
    \"eject\": \"cycle-scripts eject\"
  },
  \"babel\": {
    \"presets\": [
      \"es2015\"
    ]
  }
}"
`;
